lib/Error.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\RST;
+
+use Throwable;
+
+use function sprintf;
+
+class Error
+{
+    /** @var string */
+    private $message;
+
+    /** @var string|null */
+    private $file;
+
+    /** @var int|null */
+    private $line;
+
+    /** @var Throwable|null */
+    private $throwable;
+
+    public function __construct(string $message, ?string $file = null, ?int $line = null, ?Throwable $throwable = null)
+    {
+        $this->message   = $message;
+        $this->file      = $file;
+        $this->line      = $line;
+        $this->throwable = $throwable;
+    }
+
+    public function __toString(): string
```

Comment:
This does not seem to be used the magic way, so let's rename it?
```suggestion
    public function asString(): string
```